File: mobile/lib/features/articles/domain/entities/article.dart
import 'package:equatable/equatable.dart';

class Article extends Equatable {
  final String id;
  final String title;
  final String content;
  final String authorName;
  final DateTime createdAt;

  const Article({
    required this.id,
    required this.title,
    required this.content,
    required this.authorName,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, title, content, authorName, createdAt];
}

File: backend/Domain/article.go
package domain

import (
	"context"
	"time"
)

type Article struct {
	ID        string    `json:"id" bson:"_id"`
	Title     string    `json:"title" bson:"title"`
	Content   string    `json:"content" bson:"content"`
	AuthorID  string    `json:"author_id" bson:"author_id"`
	AuthorName string   `json:"author_name" bson:"author_name"`
	CreatedAt time.Time `json:"created_at" bson:"created_at"`
}

type ArticleRepository interface {
	Create(ctx context.Context, article *Article) error
	FetchAll(ctx context.Context) ([]*Article, error)
}
